02-cpu-configure/cpu-configure.sh

```diff
@@ -2,108 +2,95 @@
-            }\" ${BPM_EPICS_CFG_FILE} && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${TIM_RX_EPICS_CFG_FILE} \
-    "
-
-exec_cmd "TRACE" echo "Copying generated files to CPU..."
-
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    scp \
-    ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE} \
-    root@${IP}:${BPM_HALCS_CFG_FILE}
-
-exec_cmd "INFO " echo "CPU Configured Successfully!"
+hostnamectl set-hostname "$CRATE_HOSTNAME"
+sysctl kernel.hostname="$CRATE_HOSTNAME"
+sysctl -w kernel.hostname="$CRATE_HOSTNAME"
+
+apt-get update -y
+apt-get upgrade -y
+apt-get dist-upgrade -y
+apt-get install build-essential vim tmux git rsync sshfs telnet screen picocom earlyoom procserv arch-install-scripts zabbix-agent2 tcpdump gdb strace wget lsb-release iptables iptables-persistent systemd-timesyncd -y
+apt-get autoremove -y
+
+iptables -t nat -A PREROUTING -p udp ! -s 127.0.0.1/24 --dport 5064 -j DNAT --to 255.255.255.255:5064
+iptables-save > /etc/iptables/rules.v4
+
+usermod -a lnls-bpm -G dialout
+usermod -a lnls-bpm -G systemd-journal
+useradd iocs -m --no-user-group --shell /usr/sbin/nologin --home-dir /opt/container-iocs
+
+loginctl enable-linger lnls-bpm
+loginctl enable-linger iocs
+
+cd /tmp
+wget https://packages.wazuh.com/4.x/apt/pool/main/w/wazuh-agent/wazuh-agent_4.7.5-1_amd64.deb && sudo WAZUH_MANAGER='wazuh.cnpem.br' WAZUH_AGENT_GROUP='LNLS' dpkg -i ./wazuh-agent_4.7.5-1_amd64.deb
+
+sed -i '/Server=127.0.0.1/c\Server=zabbix.lnls.br' /etc/zabbix/zabbix_agent2.conf
+sed -i '/ServerActive=127.0.0.1/c\ServerActive=zabbix.lnls.br' /etc/zabbix/zabbix_agent2.conf
+
+sed -i '/GRUB_CMDLINE_LINUX=""/c\GRUB_CMDLINE_LINUX="pci=noaer intel_iommu=on console=ttyS1,115200"' /etc/default/grub
+update-grub
+
+ROOTFS_PART=`findmnt -n -r / | cut -d ' '  -f 2 | cut -d '[' -f 1`
+mount $ROOTFS_PART /mnt/
+btrfs subvolume create /mnt/@snapshots
+btrfs subvolume create /mnt/@home
+mv /home/lnls-bpm /mnt/@home
+mkdir /mnt/@snapshots/rootfs
+mkdir /mnt/@snapshots/home
+umount /mnt
+mkdir /snapshots
+mount $ROOTFS_PART /home -o subvol=/@home
+mount $ROOTFS_PART /snapshots -o subvol=/@snapshots
+genfstab -U / > /etc/fstab
+
+printf "\ndeb http://deb.debian.org/debian/ bookworm-backports main non-free-firmware contrib non-free\n" >> /etc/apt/sources.list
+apt-get update
+apt-get install podman podman-compose -t bookworm-backports -y
+
+printf '#/bin/sh\n\nset -e\nsnap_root_path=/snapshots/rootfs/rootfs-$(date +%%F-%%H-%%M-%%S)\nsnap_home_path=/snapshots/home/home-$(date +%%F-%%H-%%M-%%S)\n\n[ ! -e "$snap_root_path" ] && btrfs subvolume snapshot -r / "$snap_root_path"\n[ ! -e "$snap_home_path" ] && btrfs subvolume snapshot -r /home "$snap_home_path"\n' > /usr/local/bin/snapshot
+chmod +x /usr/local/bin/snapshot
+
+printf "#/bin/sh\nlspci  -mm -v | grep -e PhySlot  | tr '\\\t' ' ' | sort -t' ' -nk2\n" > /usr/local/bin/pcie-list-slots
+chmod +x /usr/local/bin/pcie-list-slots
```

Comment:
Both scripts could use heredocs in order to simplify reading and editing the scripts, IMO. It's more relevant for the first script.

Reply:
Now readable.